satpy/writers/cf_writer.py

```diff
@@ -71,26 +71,42 @@
 
     >>> my_encoding = {
     ...    'my_dataset_1': {
-    ...        'zlib': True,
+    ...        'compression': 'zlib',
     ...        'complevel': 9,
     ...        'scale_factor': 0.01,
     ...        'add_offset': 100,
     ...        'dtype': np.int16
     ...     },
     ...    'my_dataset_2': {
-    ...        'zlib': False
+    ...        'compression': None,
+    ...        'dtype': np.float64
     ...     }
     ... }
     >>> scn.save_datasets(writer='cf', filename='encoding_test.nc', encoding=my_encoding)
 
 
 2) Via the ``encoding`` attribute of the datasets in a scene. For example
 
-    >>> scn['my_dataset'].encoding = {'zlib': False}
+    >>> scn['my_dataset'].encoding = {'compression': 'zlib'}
```

Comment:
Is this equivalent to `scn['my_dataset'].attrs['encoding'] = {...}`? We probably shouldn't be setting attributes by `.attrname`.

Reply:
No that's not equivalent. `data_array.attrs["encoding"]` is just an ordinary attribute for xarray and `to_netcdf` fails because that dictionary cannot be serialized. In the documentation they use `.encoding`: https://docs.xarray.dev/en/stable/user-guide/io.html#reading-encoded-data

Reply:
Ah sounds good. I wasn't sure if `encoding` was *in* the attrs or its own separate thing. It is its own separate thing then. Sounds good.